### PR TITLE
[anyrpc] Add MinGW support

### PR DESCRIPTION
**Pull request by m-kuhn**
Ports https://github.com/sgieseking/anyrpc/pull/46

- #### What does your PR fix?
  Fixes MinGW support for anyrpc

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all Yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

File: ports/anyrpc/mingw.patch [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
++++ b/src/CMakeLists.txt	2022-07-19 19:35:41.697935374 +0200
+@@ -57,6 +57,12 @@
+ # Need the winsock library for Windows
+ if (WIN32)
+     target_link_libraries(anyrpc ws2_32)
++
++    target_compile_definitions(anyrpc 
++        PRIVATE
++            WINVER=0x0A00
++            _WIN32_WINNT=0x0A00
```
Comment (FrankXie05) [inline ports/anyrpc/mingw.patch:140]:
```suggestion
+            WINVER=0x0A00
+            _WIN32_WINNT=0x0A00
```
As far as I know, the variable of this macro is for windows10, can you explain it in detail?  :)

Comment (m-kuhn) [reply ports/anyrpc/mingw.patch:140]: I am not upstream maintainer so I cannot tell.
What I can tell is it fixes mingw.

**Comment by longnguyen2004** [comment]
Could probably add a note to remove the patch once https://github.com/sgieseking/anyrpc/pull/46 is merged, otherwise LGTM!

**Review by longnguyen2004 — DISMISSED** [review]

File: ports/anyrpc/portfile.cmake [hunk 2]
```diff
@@ -5,6 +5,8 @@ vcpkg_from_github(
     SHA512 8c674d29e80ec2522d6c1ec959663958ab4e1bf1135727c3c2aaa19e62a81ddbbd1e6a46f3e4679ee02894ad2ab26e70ca7e1e6c8750f3289994311069221b53
     HEAD_REF master
     FILE_DISAMBIGUATOR 1
+    PATCHES
+        mingw.patch
```
Comment (FrankXie05) [inline ports/anyrpc/portfile.cmake:9]:
```suggestion
        mingw.patch #Remove this patch in the next version 
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/anyrpc/mingw.patch (hunk 1 above)
Comment (FrankXie05) [reply ports/anyrpc/mingw.patch:140]: Thanks for your reply, we accept this patch. We should add a note saying: Remove this patch in the next version

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
I want to give upstream an opportunity to merge (or not) https://github.com/sgieseking/anyrpc/pull/46 before merging this.

**Comment by m-kuhn** [comment]
How long do we want to wait for upstream to merge (or not)?

**Comment by FrankXie05** [comment]
@m-kuhn At least a month, we may merge this pr if upstream doesn't respond for a long time. :)

**Comment by FrankXie05** [comment]
@BillyONeal Can we merge this pr, due to upstream not responding for a long time.

**Comment by longnguyen2004** [comment]
Just merge this, since the purpose of a patch is to add something upstream doesn't have anyway

**Comment by BillyONeal** [comment]
> How long do we want to wait for upstream to merge (or not)?

It's been a month, that's enough for me. :)

Moreover, upstream indicated that they don't have serious problems with the direction in https://github.com/sgieseking/anyrpc/pull/46#issuecomment-1191687257

Thanks for your contribution!

**Pull request merged**